pkg/evs/services/snapshots.go

```diff
@@ -14,3 +21,95 @@ func GetSnapshot(c *config.CloudCredentials, id string) (*snapshots.Snapshot, er
 
 	return snapshots.Get(client, id).Extract()
 }
+
+func ListSnapshots(c *config.CloudCredentials, opts snapshots.ListOpts) ([]snapshots.Snapshot, error) {
+	client, err := getEvsV2Client(c)
+	if err != nil {
+		return nil, err
+	}
+	page, err := snapshots.ListPage(client, opts)
+	if err != nil {
+		return nil, err
+	}
+	return page.Snapshots, nil
+}
+
+func CreateSnapshotToCompleted(credentials *config.CloudCredentials, name string, volumeId string) (
+	*csi.CreateSnapshotResponse, error) {
```

Comment:
service only provides functions at the service level, and the return value of constructing `*csi.CreateSnapshotResponse` should be at the controller level

Reply:
ok, it has been modified.